### PR TITLE
Props views components documentation

### PR DESCRIPTION
**Pull request by AliciadW**
@schalkventer can you please review that comments that I have added to the Components' `propTypes`. 
Please let me know if I am on the right track. As soon as I know you are happy with this I will continue with the views.

### CODE REVIEW

File: src/components/ClickWrapper/index.jsx [hunk 1]
```diff
@@ -129,9 +129,16 @@ export default ClickWrapper;
 
 
 ClickWrapper.propTypes = {
+  /** Determines whether width should be set to 100%. **/
   full: t.bool,
+  /** Nested structure that needs this wrapper **/
```
Comment (schalkventer) [inline src/components/ClickWrapper/index.jsx]: Content nested inside this component.

Comment (schalkventer) [reply src/components/ClickWrapper/index.jsx]: Can make isRequired.

File: src/components/ClickWrapper/index.jsx [hunk 2]
```diff
@@ -129,9 +129,16 @@ export default ClickWrapper;
...
   children: t.node.isRequired,
+  /** URL that should be added to the click action.
```
Comment (schalkventer) [inline src/components/ClickWrapper/index.jsx]: 'This should be the URL to where the user will be directed when clicking on the link. Alternatively, if a function is passed that function will be executed when the user clicks on the link'

File: src/components/ClickWrapper/index.jsx [hunk 3]
```diff
@@ -143,8 +150,12 @@ ClickWrapper.defaultProps = {
 
 
 CallbackWrapper.propTypes = {
+  /** Nested structure that needs this wrapper **/
```
Comment (schalkventer) [inline src/components/ClickWrapper/index.jsx]: 'React content nested inside this component.'

File: src/components/ClickWrapper/index.jsx [hunk 4]
```diff
@@ -143,8 +150,12 @@ ClickWrapper.defaultProps = {
 
 
 CallbackWrapper.propTypes = {
```
Comment (schalkventer) [inline src/components/ClickWrapper/index.jsx]: I think we only need to add descriptions on the root components inside `index.jsx` and `Markup.jsx` files.

File: src/components/Layout/Header/index.jsx [hunk 5]
```diff
@@ -67,9 +67,15 @@ export default Header;
 
 
 Header.propTypes = {
+  /** The title that will be displayed in the header. **/
```
Comment (schalkventer) [inline src/components/Layout/Header/index.jsx]: 'The title that will be displayed in the page header-bar.'

File: src/components/Layout/Header/index.jsx [hunk 6]
```diff
@@ -67,9 +67,15 @@ export default Header;
...
   title: t.string,
+  /** Determines whether or not the back button should be displayed in the header. **/
```
Comment (schalkventer) [inline src/components/Layout/Header/index.jsx]: 'If a value is passed to this prop then a back button will be displayed in the header. If the value is a string then a user will go to that URL if clicked, if the value is a function then the function will be called when the back button is clicked'.

File: src/components/Layout/Header/index.jsx [hunk 7]
```diff
@@ -67,9 +67,15 @@ export default Header;
...
   back: t.oneOfType([t.string, t.func]),
+  /** Whether a custom Link component should be used to resolve links,
+   * for example the Link component from React Router or Gatsby.
+   * If not supplied, card link will just fall back to a regular <a> tag. */
   link: t.node,
+  /** Displays the language that has been selected by the user. **/
```
Comment (schalkventer) [inline src/components/Layout/Header/index.jsx]: 'Displays the language that has been selected by the user (as isoKey).'

File: src/components/Layout/index.jsx [hunk 8]
```diff
@@ -53,12 +53,19 @@ export default Layout;
 
 
 Layout.propTypes = {
+  /** Nested structure that needs this wrapper **/
```
Comment (schalkventer) [inline src/components/Layout/index.jsx]: 'React content nested inside this component.'

File: src/components/Layout/index.jsx [hunk 9]
```diff
@@ -53,12 +53,19 @@ export default Layout;
...
   children: t.node.isRequired,
+  /** The title that will be displayed in the header. **/
```
Comment (schalkventer) [inline src/components/Layout/index.jsx]: 'The title that will be displayed in the page header-bar.'

File: src/components/Layout/index.jsx [hunk 10]
```diff
@@ -53,12 +53,19 @@ export default Layout;
...
   title: t.string,
+  /** Determines whether or not the back button should be displayed in the header. **/
```
Comment (schalkventer) [inline src/components/Layout/index.jsx]: 'If a value is passed to this prop then a back button will be displayed in the header. If the value is a string then a user will go to that URL if clicked, if the value is a function then the function will be called when the back button is clicked'.

File: src/components/Layout/index.jsx [hunk 11]
```diff
@@ -53,12 +53,19 @@ export default Layout;
...
   back: t.oneOfType([t.string, t.func]),
+  /** Determines whether or not the header should be displayed on a page. **/
   header: t.bool,
+  /** Determines whether or not the footer should be displayed on a page. **/
   footer: t.bool,
+  /** Displays the language that has been selected by the user. **/
```
Comment (schalkventer) [inline src/components/Layout/index.jsx]: 'Displays the language that has been selected by the user (as isoKey) in the header-bar. Is also used to determine what language should be used for the hardcode UI-specific/instructional text on the page.'

File: src/components/Layout/index.jsx (hunk 8 above)
Comment (AliciadW) [reply src/components/Layout/index.jsx]: Updated

File: src/components/Layout/index.jsx (hunk 9 above)
Comment (AliciadW) [reply src/components/Layout/index.jsx]: Updated

File: src/components/Layout/index.jsx (hunk 10 above)
Comment (AliciadW) [reply src/components/Layout/index.jsx]: Updated

File: src/components/Layout/index.jsx (hunk 11 above)
Comment (AliciadW) [reply src/components/Layout/index.jsx]: Updated

File: src/components/ClickWrapper/index.jsx (hunk 1 above)
Comment (AliciadW) [reply src/components/ClickWrapper/index.jsx]: Updated

File: src/components/ClickWrapper/index.jsx (hunk 4 above)
Comment (AliciadW) [reply src/components/ClickWrapper/index.jsx]: Makes sense, updated as requested

File: src/components/ClickWrapper/index.jsx (hunk 2 above)
Comment (AliciadW) [reply src/components/ClickWrapper/index.jsx]: Updated

**Comment by AliciadW** [comment]
@schalkventer can we close this PR? Assuming this is not needed anymore.

**Comment by AliciadW** [comment]
Too old

**Pull request closed without merging**